Fractional pixel 1D extraction for NIRSpec IFU

jdavies-st opened the pull request:
This is a notebook describing how to do fractional pixel extraction with a variable-radius circular aperture on NIRSpec IFU cubes.  It incorporates the use of `glue-jupyter` as well.

@eteq @PatrickOgle

PatrickOgle commented:
This is just a concept notebook, to demonstrate two things:
1) How to get region parameters from cubeViz
2) How to extract a spectrum with an expanding aperture.

Both will actually appear in my IFU cube modeling notebook in jdat_notebooks eventually.
This notebook should be moved to either dat_pyinthesky/concept_APIs or jdaviz/notebooks/concepts

jdavies-st commented:
I've moved the notebook to `dat_pyinthesky/concept_APIs`, provided some explanatory text, moved some of the inline comments to markdown cells, and deleted extra cells.

eteq commented:
Note the CI is not a concern here because the CI only runs on things in `jdat_notebooks`, not the concept APIs.  concepts are intentionally not run because some of them are more pseudo-code or "code should go here" than actually runnable notebooks.  So I'm merging now, thanks @jdavies-st !
